src/main/java/practice/StreamPractice.java

```diff
@@ -59,7 +85,14 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex() == Person.Sex.WOMAN
+                        && p.getCats().size() != 0
+                        && p.getAge() >= femaleAge)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
+                .map(Cat::getName)
```

Comment:
You can combine these three lines into one as such:
```suggestion
.flatMap(p -> p.getCats().stream().map(Cat::getName))
```